cognite/neat/graph/extractors/_asset.py

```diff
@@ -0,0 +1,83 @@
+from datetime import datetime
+
+import pytz
+from cognite.client import CogniteClient
+from cognite.client.data_classes import Asset
+from rdflib import RDF, Literal, Namespace
+
+from cognite.neat.constants import DEFAULT_NAMESPACE
+from cognite.neat.graph.models import Triple
+from cognite.neat.utils.utils import string_to_ideal_type
+
+from ._base import BaseExtractor
+
+
+class AssetExtractor(BaseExtractor):
+    def __init__(self, client: CogniteClient, data_set_id: int | None = None, namespace: Namespace | None = None):
+        self.namespace = namespace or DEFAULT_NAMESPACE
+        self.client = client
+        self.dataset_id = data_set_id
+
+    def extract(self, limit: int = -1) -> list[Triple]:
+        """Extracts an asset with the given asset_id."""
+
+        triples: list[Triple] = []
+        for asset in self.client.assets.list(limit=limit, data_set_ids=self.dataset_id):
+            triples.extend(self._asset2triples(asset, self.namespace))
+
+        return triples
```

Comment:
Suggest replacing the above with this:
```suggestion
    def __init__(self, assets: Iterable[Asset], namespace: Namespace | None = None):
        self.namespace = namespace or DEFAULT_NAMESPACE
        self.assets = assets

    @classmethod
    def from_dataset(cls, client: CogniteClient, data_set_external_id: str, namespace: Namespace | None = None):
        return cls(client.assets(data_set_external_id=data_set_external_id), namespace)

    @classmethod
    def from_hierarchy(cls, client: CogniteClient, root_asset_external_id: str, namespace: Namespace | None = None):
        return cls(client.assets(asset_subtree_external_ids=root_asset_external_id), namespace)
        
    @classmethod
    def from_file(cls, file_path: str, namespace: Namespace | None = None):
        return cls(AssetList.load(Path(file_path).read_text()), namespace)

    def extract(self) -> Iterable[Triple]:
        """Extracts an asset with the given asset_id."""

        for asset in self.assets:
            yield self._asset2triples(asset, self.namespace)
```

This has multiple advantages:
* We can support different ways of retrieving the assets, from a dataset or root asset as shown above, or from file.
* Note that the `__call__` method is an iterator, which means instead of downloading all assets into memory, it is doing it only one chunk at a time. Thus it is very memory efficient.
* Equally making the `.extract` method ensure that we do not have to keep all triples in memory.

This also simplifies the test to no longer require mocking.

```pythoyn
def test_asset_extractor():
    assets = AssetList.load((ASSET_EXTRACTOR_DATA / "nordic44_assets.yaml").read_text())

    triples = list(AssetExtractor(assets).extract())

    assert len(triples) == 18675
    assert sum(1 for t in triples if t[2] == URIRef("http://purl.org/cognite/neat#Substation")) == 44
```